Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -912,12 +923,17 @@ def resource_version(
         puts if debug && strategy.blank?
         next if strategy.blank?
 
-        strategy_data = strategy.find_versions(
+        strategy_args = {
           url:           url,
-          regex:         livecheck_regex,
           homebrew_curl: false,
-          &livecheck_strategy_block
-        )
+        }
+        strategy_args[:regex] = livecheck_regex if livecheck_regex.present?
+
+        strategy_data = if livecheck_strategy_block.present?
+          strategy.find_versions(**strategy_args, &livecheck_strategy_block)
```

Comment:
Does this not work if the block is `nil`?

Reply:
Testing this again, `strategy_data = strategy.find_versions(**strategy_args, &livecheck_strategy_block)` seems fine 👍